### PR TITLE
chore: remove the `tslint.json` file

### PR DESCRIPTION
**Pull request by mrazauskas**
This PR removes the `tslint.json` file.

This file holds configuration / rules for `tslint`. It is not used after #1979. (Or?)

### CODE REVIEW

File: type-definitions/ts-tests/tslint.json [hunk 1]
```diff
@@ -1,10 +0,0 @@
-{
-  "extends": "dtslint/dtslint.json",
```
Comment (mrazauskas) [inline type-definitions/ts-tests/tslint.json:2]: Despite having `dtslint` in its name, this file simply holds a list of `tslint` rules, reference: https://github.com/microsoft/dtslint/blob/master/dtslint.json

**Comment by mrazauskas** [comment]
Ah.. `dtslint` is using it. So this could be part of #1986 (or land as a follow up).

/cc @arnfaldur

**Comment by arnfaldur** [comment]
Indeed, removing the file has no effect on the test suite after the changes in https://github.com/immutable-js/immutable-js/pull/1986

**Comment by jdeniau** [comment]
Closing this, as it does not work.

**Pull request closed without merging**